### ISSUE TITLE
mpirun check failed with version 

### ISSUE THREAD

**Issue by wjxiz1992** [horovod/horovod#3380]
**Environment:**
4. MPI version: 4.1.2a1

**Bug report:**
I think it's not a Horovod problem but still report here. 
With MPI 4.1.2a1 version, I see 

```
Checking whether extension tensorflow was built with MPI.
Extension tensorflow was built with MPI.
Was unable to run mpirun --version:
mpirun: Error: unknown option "--version"
```
I tried to modify Horovod to change it to "-version" ,"-V", all failed with same `unknown option` error.
Then I go back to version 4.0.5, everything is fine.

**Comment by maxhgerlach** [comment]
Pretty sure your OpenMPI install was broken. `mpirun` should still support a `--version` argument: https://www.open-mpi.org/doc/current/man1/mpirun.1.php

Feel free to reopen if this is specific to Horovod.

**Comment by wjxiz1992** [comment]
Hi, you mean you can run horovod successfully with mpi version 4.1.2? I was
reporting this because: I can get mpirun ––version without error from the
shell if I manually log into my running container while it only fails
inside Horovod.

**Comment by maxhgerlach** [comment]
Well, how should I have inferred that? You wrote yourself
>  I think it's not a Horovod problem but still report here.

**Comment by wjxiz1992** [comment]
Yeah, as I am using the same Horovod version, the only difference is the
mpi version. My intention is only to inform any Horovod user that get the
same error.

Btw, do you know any mpi version that does not support the ––version
option? It’s so weird that mpirun executable is founded but report such
error.

**Comment by maxhgerlach** [comment]
I don't know about `--version`, but for instance here's an issue where mpirun complained about `unknown option "-np"`, which happened to be the first on the command line: https://github.com/open-mpi/ompi/issues/4557 It's not a very helpful message, but there the problem was related to a broken installation.

**Comment by maxhgerlach** [comment]
(reopening because it's not clear that this is unrelated to Horovod)
